parsing/lexer.mll

```diff
@@ -861,6 +893,27 @@ and directive already_consumed = parse
             update_loc lexbuf (Some name) (line_num - 1) true 0;
             token lexbuf
       }
+  (* Lexer rule for the #syntax directive *)
+  | "syntax" [' ' '\t']+ (lowercase identchar* as mode) [' ' '\t']*
```

Comment:
Maybe make it be `+` at the end? We require there to be some whitespace between the mode and the toggle, and this makes that clearer.